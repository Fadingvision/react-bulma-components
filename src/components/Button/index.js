import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

function Button({
  size,
  type,
  className,
  component: Component,
  fullWidth,
  loading,
  style,
  children,
  ...other
}) {
  const classnames = classNames(
    'button',
    {
      [`is-${size}`]: size,
      [`is-${type}`]: type,
      'is-loading': loading
    },
    className
  );

  const fullWidthStyle = {
    width: '100%'
  };

  let conditionStyle;
  if (fullWidth) {
    conditionStyle = style ? Object.assign(fullWidthStyle, style) : fullWidthStyle;
  }

  return (
    <Component className={classnames} style={conditionStyle} {...other}>
      {children}
    </Component>
  );
}

Button.defaultProps = {
  loading: false,
  fullWidth: false,
  disabled: false,
  component: 'button'
};

Button.PropTypes = {
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  type: PropTypes.oneOf([
    'white',
    'light',
    'dark',
    'black',
    'link',
    'primary',
    'info',
    'success',
    'warning',
    'danger',
  ]),
  onClick: PropTypes.func,
  loading: PropTypes.bool,
  component: PropTypes.string,
};

export default Button;
