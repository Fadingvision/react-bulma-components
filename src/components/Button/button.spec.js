import React, { Component } from 'react';
import { shallow, mount, render } from 'enzyme';
import Button from './index';

describe('Button test', () => {

  it('should render correctly', () => {
    const wrapper = render(<Button>button text</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should change loading state instantly by default', () => {
    class DefaultButton extends Component {
      state = {
        loading: false,
      };
      enterLoading = () => {
        this.setState({ loading: true });
      }
      render() {
        return <Button loading={this.state.loading} onClick={this.enterLoading}>Button</Button>;
      }
    }
    const wrapper = mount(
      <DefaultButton />
    );
    wrapper.simulate('click');
    expect(wrapper.hasClass('is-loading')).toBe(true);
  });
});
