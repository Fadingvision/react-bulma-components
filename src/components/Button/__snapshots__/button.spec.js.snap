// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test should render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "button",
        },
        "children": Array [
          Object {
            "data": "button text",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
